src/atmi-backward-compat.cmake

```diff
@@ -0,0 +1,130 @@
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+cmake_minimum_required(VERSION 3.16.8)
+
+set(ATMI_BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR})
+set(ATMI_WRAPPER_DIR ${ATMI_BUILD_DIR}/wrapper_dir)
+set(ATMI_WRAPPER_INC_DIR ${ATMI_WRAPPER_DIR}/include)
+set(ATMI_WRAPPER_BIN_DIR ${ATMI_WRAPPER_DIR}/bin)
+if(CMAKE_BUILD_TYPE MATCHES Debug)
+  set(ATMI_LIB_NAME "lib-debug")
+else()
+  set(ATMI_LIB_NAME "lib")
+endif()
+set(ATMI_WRAPPER_LIB_DIR ${ATMI_WRAPPER_DIR}/${ATMI_LIB_NAME})
+
+set(PUBLIC_HEADERS
+    atmi.h
+    atmi_interop_hsa.h
+    atmi_runtime.h
+  )
+
+#Function to generate header template file
+function(create_header_template)
+    file(WRITE ${ATMI_WRAPPER_DIR}/header.hpp.in "/*
+    Copyright (c) 2015 - 2021 Advanced Micro Devices, Inc. All rights reserved.
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the \"Software\"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED \"AS IS\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+   THE SOFTWARE.
+   */\n\n#ifndef @include_guard@\n#define @include_guard@ \n\n\#warning This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with atmi\n@include_statements@ \n\n#endif")
```

Comment:
Please change to #pragma message as #warning may not work in Windows

Reply:
Change #warning to #pragma message